controllers/dockercluster_controller.go

```diff
@@ -38,88 +46,131 @@ type DockerClusterReconciler struct {
 // +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=dockerclusters,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=dockerclusters/status,verbs=get;update;patch
 
-// Reconcile handles DockerCluster events
-func (r *DockerClusterReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, rerr error) {
+// Reconcile reads that state of the cluster for a DockerCluster object and makes changes based on the state read
+// and what is in the DockerCluster.Spec
+func (r *DockerClusterReconciler) Reconcile(req reconcile.Request) (_ reconcile.Result, reterr error) {
 	ctx := context.Background()
-	log := r.Log.WithValues("dockercluster", req.NamespacedName)
-	log.Info("Reconciling cluster")
+	log := log.Log.WithName(controllerName).
+		WithValues("namespace", req.Namespace, "dockerCluster", req.Name)
 
-	dockerCluster := &infrastructurev1alpha2.DockerCluster{}
+	// Fetch the DockerCluster instance
+	dockerCluster := &infrav1.DockerCluster{}
 	if err := r.Client.Get(ctx, req.NamespacedName, dockerCluster); err != nil {
 		if apierrors.IsNotFound(err) {
 			return ctrl.Result{}, nil
 		}
-		log.Error(err, "failed to get DockerCluster")
 		return ctrl.Result{}, err
 	}
 
+	log = log.WithValues("version", dockerCluster.APIVersion)
+
+	// Fetch the Cluster.
 	cluster, err := util.GetOwnerCluster(ctx, r.Client, dockerCluster.ObjectMeta)
 	if err != nil {
-		log.Error(err, "Failed to get owning cluster")
-		return ctrl.Result{}, err
+		return reconcile.Result{}, err
 	}
 	if cluster == nil {
-		log.Error(err, "Waiting for an OwnerReference to appear")
-		return ctrl.Result{RequeueAfter: 30 * time.Second}, nil
+		log.Info("Waiting for Cluster Controller to set OwnerRef on DockerCluster")
+		return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
```

Comment:
Do we need to requeue here? shouldn't we be setting up watches against the Cluster resources?

Reply:
💯. No reason to requeue this.

Reply:
I have added the watch that was not implemented and removed the RequeueAfter.
Should I send I fix to CAPA as well?

Reply:
yes, please